code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,254 @@
+	var/list/assasination = list()
+	var/list/protection = list()
+	for(var/i = 0 , i < 2 , i++)
+		var/pck = pick("assasinate","stalk","protect")
+		switch(pck)
+			if("assasinate")
+				var/datum/objective/assassinate/A = new
+				A.owner = owner
+				A.find_target(owner,protection)
+				assasination += A.target
+				objectives += A
+			if("stalk")
+				var/datum/objective/stalk/S = new
+				S.owner = owner
+				S.find_target()
+				S.update_explanation_text()
+				objectives += S
+			if("protect")
+				var/datum/objective/protect/P = new
+				P.owner = owner
+				P.find_target(owner,assasination)
+				protection += P.target
+				objectives += P
+
+	var/datum/objective/sacrifice_ecult/SE = new
+	SE.update_explanation_text()
+	objectives += SE
+
+	return
+
+/datum/antagonist/ecult/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	handle_clown_mutation(current, mob_override ? null : "Knowledge described in the book allowed you to overcome your clownish nature, allowing you to wield weaponry.")
+	current.faction |= "ecult"
+
+/datum/antagonist/ecult/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	var/mob/living/current = owner.current
+	if(mob_override)
+		current = mob_override
+	handle_clown_mutation(current, removing = FALSE)
+	current.faction -= "ecult"
+
+
+/datum/team/ecult
+	name = "Eldritch Cult"
+
+////////////////
+// Knowledge //
+////////////////
+
+/datum/antagonist/ecult/proc/gain_knowledge(datum/eldritch_knowledge/EK)
+	if(has_knowledge(EK))
+		return FALSE
+	var/datum/eldritch_knowledge/initialized_knowledge = new EK
+	researched_knowledge |= initialized_knowledge
+	initialized_knowledge.on_gain(owner.current)
+	return TRUE
+
+/datum/antagonist/ecult/proc/get_researchable_knowledge()
+	var/list/researchable_knowledge = list()
+	var/list/banned_knowledge = list()
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		researchable_knowledge |= EK.next_knowledge
+		banned_knowledge |= EK.banned_knowledge
+		banned_knowledge |= EK.type
+	researchable_knowledge -= banned_knowledge
+	return researchable_knowledge
+
+/datum/antagonist/ecult/proc/has_knowledge(datum/eldritch_knowledge/EK)
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK1 = X
+		if(initial(EK.name) == EK1.name)
+			return TRUE
+	return FALSE
+
+/datum/antagonist/ecult/proc/get_knowledge(datum/eldritch_knowledge/EK)
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK1 = X
+		if(istype(EK1,EK))
+			return EK1
+
+/datum/antagonist/ecult/proc/get_all_knowledge()
+	return researched_knowledge
+
+////////////////
+// Objectives //
+////////////////
+
+/datum/objective/stalk
+	name = "spendtime"
+	var/timer = 1800 //5 minutes
+
+/datum/objective/stalk/update_explanation_text()
+	if(timer == initial(timer))//just so admins can mess with it
+		timer += pick(-600, 600)
```

Comment:
if this timer increases to be greater than the default amount of time, then decrements down to 1800 on the same tick that update_explanation_text() gets called, will the amount of time left on the timer be randomized again?

Reply:
yes, but what admin would make set the timer to exactly 1801.

Reply:
well, the timer's gonna be decrementing while you're near your target, yeah?

Reply:
update_explanation_text() doesn't fire every tick, but once during the objective setup stage.